hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -989,7 +989,9 @@ def check_result(result):
             return result
 
         try:
-            return check_result(self.__data_cache[buffer])
+            cached = check_result(self.__data_cache[buffer])
+            if cached.status > Status.OVERRUN or extend == 0:
+                return cached
```

Comment:
If I'm understanding correctly, the idea here is that:

- For non-extend calls, any cached result is OK to return.
- For extend calls that aren't overrun, the test ends before extend would make any difference, so the cached result is OK to return.

In the extend+overrun case, the cached overrun is only valid for the actual buffer on its own, and doesn't tell us whether buffer+extend would overrun, so we have to ignore the cached result and actually simulate/run the test.

Reply:
Correct, yes.